Converted NavBar to MUI components

dyazdani opened the pull request:
Closes #72 

Not logged in:
![Screen Shot 2023-11-27 at 15 56 07](https://github.com/dyazdani/zoop/assets/99094815/557e6d2a-bbc6-4fae-82df-a90b6a549b4b)

Logged in:
![Screen Shot 2023-11-27 at 15 56 42](https://github.com/dyazdani/zoop/assets/99094815/d28d18fe-dd29-4dd3-a500-dde9e5a0771c)

paloobi commented on client/components/Nav.tsx:
```diff
@@ -1,16 +1,167 @@
+    const [accountCircleAnchorEl, setAccountCircleAnchorEl] = useState<null | HTMLElement>(null);
+
+    const navigate = useNavigate();
+    const isLoggedIn = !!selectCurrentUser;
```
This will always be false, because currently the authSlice is not storing the user after register/login. Might be worth filing a bug to get that fixed in a separate PR. But you won't be able to test this until logged in user is properly loaded into the store

dyazdani replied:
I think we added `state.user` to the send a zoop feature PR right?

paloobi commented on client/components/Nav.tsx:
```diff
@@ -1,16 +1,167 @@
+                                navigate("/");
+                            }}
+                        > 
+                            Home
```
tbh I tend to recommend instead of having an explicit link to home, that you just link to home when the header is clicked. It's a common design pattern on websites, and calling it Home in a user-facing link like this feels a bit old-fashioned to me

paloobi commented on client/components/Nav.tsx:
```diff
@@ -1,16 +1,167 @@
+                        }}
+                    >
+                        <Typography 
+                            variant="h6" 
```
I think you need to set this as an h1 - it is an h1 because it represents the top-level header on this app. You can set custom styles to change the font size if you want, but I don't recommend setting it as an h6, it will lead to a poor experience for users on screen readers

paloobi commented on client/components/Nav.tsx:
```diff
@@ -1,16 +1,167 @@
+                        >
+                            ZOOP
+                        </Typography>
+                        <Typography
+                            variant="h6"
+                            component="div"
+                        >
```
This is a purely presentational element, not a real header. You might want to add an aria label to indicate that it's just for presentation (I forget which label that is), otherwise it may be read out loud as "dash dash dash" in screen readers

dyazdani replied:
Hmm. There is no aria label prop for the Typography component. I've been wondering, Should the Zoop header with the paper airplane be a png logo instead?

dyazdani replied:
I could wrap the Typography component in a Box that has "div" for its component prop and give that and ariaLabel of "presentation" or "none" (those are the aria-label attribute values for this situation, according to MDN).

dyazdani replied:
Never mind. When I set the `Typography` component prop to `div` it has an aria-label prop.

dyazdani replied:
> I've been wondering, Should the Zoop header with the paper airplane be a png logo instead?

@paloobi Ashley drew a couple of quick svg images for Zoop. I'll share them in the main comments.

paloobi reviewed (changes requested):
The one piece of CSS you will need in this app is you will need to remove the padding from the body. In your main CSS file (is there one in App.css or is there an index.css?) you should remove the padding so that MUI header fills the whole screen

RC for:
- remove padding on body
- use h1 for top-level heading
- make the dashes presentational so they aren't read by screen readers
- make sure the user is loaded into the store on register/login

paloobi commented:
Honestly, the more I think about it, the more I think it might be wise to just eliminate the Icon button on the left with the menu icon? I can't picture what would go in there. It wasn't in the original design, was it?

dyazdani commented:
> The one piece of CSS you will need in this app is you will need to remove the padding from the body. In your main CSS file (is there one in App.css or is there an index.css?) you should remove the padding so that MUI header fills the whole screen

There is a `style.css` file in the `public` folder, but no `App.css` or `index.css`. Should I take away the padding using `style.css`?

dyazdani commented:
@paloobi @jvaneyken For fun, Ashley drew us some rough logo svg files in case we want to use them for Zoop. The typeface for the Zoop logo with text can be changed easily. The logo without text is meant to be at the top of a Zoop component.

Let me know what you guys think about using some version of these instead of the header with text dashes and a send icon. MUI has an SvgIcon component for custom SVGs. 

For the app's header:
![Zoop_planeandletters](https://github.com/dyazdani/zoop/assets/99094815/d95df526-d10c-4144-8477-5022c91e8ce6)

For the top of a Zoop component:
![Zoop_justplane](https://github.com/dyazdani/zoop/assets/99094815/b1d7f04e-8e32-42e8-a68c-a39728171ee8)

paloobi commented:
I really love both of those logos!!! You may need to make the paper airplane more white so it stands out well on the blue background? Have you tested the logo over blue? Only issue would be contrast.

Overall love em and think they're great, lots of personality :)

paloobi commented:
@dyazdani  also yes, using `style.css` should be fine for removing the padding from body.

dyazdani commented:
I'll see if Ashley can change it to white and see what it looks like on a blue background. Then I'll share the logos again.

dyazdani commented:
@paloobi Do you have experience getting svg and png files to render in a React app that's using webpack and TS? I've been at it trying to get the Zoop logo to render, but I keep getting this error:
![Screen Shot 2023-11-22 at 01 53 57](https://github.com/dyazdani/zoop/assets/99094815/98ecd02f-b006-4463-b693-31ac3b8ca644)

I've tried a handful of webpack loaders with their respective configurations to get webpack to recognize these image files, but no luck thus far. Any tips would be much appreciated. Thanks!

paloobi commented:
@dyazdani it might help if you upload the latest changes? I don't know how you're referencing the svg.

I don't have a ton of experience with webpack loaders so for that piece you would have as much luck as me searching around. But I can help look around a bit too once I see what you're trying to do

dyazdani commented:
Are they supposed to in public? I haven't tried that. Would I have them in the public folder or in an assets folder in public?

paloobi commented:
Usually I've seen an images folder in public

paloobi commented:
Have you looked into using webpack to transform them into online? https://www.pluralsight.com/guides/how-to-load-svg-with-react-and-webpack

dyazdani commented:
> Have you looked into using webpack to transform them into online? https://www.pluralsight.com/guides/how-to-load-svg-with-react-and-webpack

Is this tutorial factoring in Typescript? I've tried a bunch of loaders for webpack but none of them seem to work. Not sure if I tried this specific one or not.

dyazdani commented:
> I don't know that in this app you would necessarily need to load it with TS/webpack anyway? Wouldn't the images go in the /public directory?

I tried importing the file from the public directory and I still get the same error. I'm not sure what else to try at this point. It seems odd that I can't use an svg or png file.

dyazdani commented:
The problem with image import is two-fold. First, apparently because I am using webpack and TS, I have to make declarations in order for the app to recognize PNG or SVG file types. Currently I am getting this error:
![Screen Shot 2023-11-26 at 11 13 08](https://github.com/dyazdani/zoop/assets/99094815/9b2232ec-5a5b-4dc2-888d-6f626843a7d5)

I have created a file with the following declaration: `declare module "*.png";` and put it in a file called `declarations.d.ts` and put it in the root directory of the project. That has not worked so I moved the file to the public, src, and client folders; I tried giving the file different names (`index.d.ts`, `custom.d.ts`). Every time I have changed the `tsconfig.json` file's `"include"` property to include the file, whatever name it has or directory it is in. Nothing has worked to eliminate that error.

Additionally, the console logs an error when I try to run the app:
![Screen Shot 2023-11-26 at 11 33 44](https://github.com/dyazdani/zoop/assets/99094815/8bf46959-7628-4135-9227-d8265de9f7f2)

It seems that webpack does not currently recognize the file. I have webpack 5, so I followed this tutorial, but it still did not work: https://webpack.js.org/guides/asset-modules/

I need help. Maybe we just scrap the nice logo if this keeps blocking us for too long. But it would be nice to get an image in the app. Seems crazy that it's this difficult (unless I am messing something up, which is likely).

dyazdani commented:
Committed the img tag solution we went over today.

paloobi commented on client/components/AppHeader.tsx:
```diff
@@ -0,0 +1,129 @@
+                    )}
+                </Box>
+            </Stack>
+            {/* TODO: move this to App.tsx */}
+            <NavLink to="/">Home</NavLink>
+            <NavLink to="/login">Login</NavLink>
+            <NavLink to="/register">Register</NavLink>
+            <NavLink to="/me">Me</NavLink>
```
do you still need this? couldn't it be deleted now that you have the navbar?

dyazdani replied:
Indeed this was just here for dev purposes before the navbar was done. I'll remove it.

paloobi reviewed (changes requested):
I think you can remove the old nav links now, right?

Also probably would be worth pulling from main to add the SendZoop button

paloobi commented on client/features/authSlice.tsx:
```diff
@@ -28,13 +28,21 @@ const authSlice = createSlice({
         state.user = payload.user
+=======
+        state.user = payload.user;
+>>>>>>> main
```
Resolve/remove these merge conflicts

dyazdani replied:
VS Code and GitHub didn't alert me to this, but it was there when I opened the file. Thanks for catching it!

dyazdani commented:
@paloobi It's saying there is merge conflict with `bundle.js`. Eek! It should be in the gitignore. Should I resolve this merge conflict through VS Code's editor or is there a better way to sort this out?

dyazdani commented:
I did that. I don't see bundle.js in this branch on Github anymore. I'll have to remove it from `main` too I suppose. Is that what you meant?
